include/xtensor-python/xtensor_type_caster_base.hpp

```diff
@@ -113,6 +123,44 @@ namespace pybind11
             }
         };
 
+        template <class T, class FSH, xt::layout_type L>
+        struct pybind_array_dim_checker<xt::xtensor_fixed<T, FSH, L>>
+        {
+            template <class B>
+            static bool run(const B& buf)
+            {
+                return buf.ndim() == FSH::size();
+            }
+        };
+
+
+        template <class T>
+        struct pybind_array_shape_checker
+        {
+            template <class B>
+            static bool run(const B& buf)
+            {
+                return true;
+            }
+        };
+
+        template <class T, class FSH, xt::layout_type L>
+        struct pybind_array_shape_checker<xt::xtensor_fixed<T, FSH, L>>
+        {
+            template <class B>
+            static bool run(const B& buf)
+            {
+                auto shape = FSH();
+                for (std::size_t i = 0; i < shape.size(); ++i)
```

Comment:
This could be simplified with `std::equal`

Reply:
I implemented the first version with `std::equal. The problem is that `buf.shape()`is not an iterator.

Reply:
Then please use `&(buf.shape(0)` and we should definitely open an issue on the pybind11 repo so that they provide an iterator API for the shape.